app/controllers/groups_controller.rb

```diff
@@ -443,20 +443,12 @@ def request_membership
         .pluck("users.username")
     )
 
-    raw = <<~EOF
-      #{reason}
-
-      ---
-      <a href="#{Discourse.base_uri}/g/#{group.name}/requests">
-        #{I18n.t('groups.request_membership_pm.handle')}
-      </a>
-    EOF
-
     post = PostCreator.new(current_user,
       title: I18n.t('groups.request_membership_pm.title', group_name: group.name),
-      raw: raw,
+      raw: params[:reason],
       archetype: Archetype.private_message,
       target_usernames: usernames.join(','),
+      custom_fields: { membership_request: group.name },
```

Comment:
What if the name of the group changes in the meantime?

Reply:
Also `membership_request` might not be the best name for that custom field? What about `requested_group_id`?